feat(#918): Update `jeo-maven-plugin` Documentation"

volodya-lombrozo opened the pull request:
In this PR I removed the outdated `Transformation Method` section, added `disassemble` example and added the comprehensive list of `jeo` objects with correspoding descriptions.

Closes: #918.

yegor256 commented on docs/jeo-objects.md:
```diff
@@ -0,0 +1,83 @@
+# Full List of Jeo Objects
```
@volodya-lombrozo why can't you put this list into the README? It's not yet too big for that.

yegor256 replied:
@volodya-lombrozo also, maybe it's better to generate this list by a script executed by GitHub Actions and inject into the README. This will guarantee it's correctness.

yegor256 reviewed (changes requested):
@volodya-lombrozo looks cool, see a few ideas above

volodya-lombrozo commented:
@yegor256 I've changed the documentation according with your suggestions. Could you have a look, please?

As for this:

> @volodya-lombrozo also, maybe it's better to generate this list by a script executed by GitHub Actions and inject into the README. This will guarantee its correctness.

I added one more puzzle.

volodya-lombrozo commented:
@yegor256 Friendly reminder

yegor256 reviewed (approved):
@volodya-lombrozo LGTM

volodya-lombrozo commented:
@rultor merge

rultor commented:
> @rultor merge

@volodya-lombrozo OK, I'll try to merge now. You can check the progress of the merge [here](https://www.rultor.com/t/46247-2582105065).

rultor commented:
> @rultor merge

@volodya-lombrozo Done! FYI, the full log is [here](https://www.rultor.com/t/46247-2582105065) (took me 16min).

0crat commented:
@volodya-lombrozo Great job on your contribution! You've earned +4 points: +4 base, +16 for 473 hits-of-code (max), -8 for exceeding 200 hits-of-code, and -8 for going over our recommended size. Remember, [smaller contributions](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) often lead to better rewards. Keep coding! Your current balance is [-2](https://www.eolang.org/zerocracy/objectionary-vitals.html).

<!-- Thanks for the contribution! You've earned +4 points for this: +4 as a basis; -8 for too many hits-of-code (473 >= 200); +8 to give you at least something. Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running balance is [-2](https://www.eolang.org/zerocracy/objectionary-vitals.html).  -->

0crat commented:
@yegor256 Great job on the review! 🌟 You've earned +4 points, bringing your balance to [-52](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep up the excellent work and watch your score grow!

<!-- Thanks for the review! You've earned +4 points.  Your running balance is [-52](https://www.eolang.org/zerocracy/objectionary-vitals.html).  -->
